evol/problems/tsp.py

```diff
@@ -0,0 +1,74 @@
+import math
+from typing import List, Union
+
+from .problem import Problem
+
+
+def _rotating_window(arr):
+    for i, city in enumerate(arr):
+        yield arr[i-1], arr[i]
+
+
+class TSPProblem(Problem):
+    def __init__(self, distance_matrix):
+        self.distance_matrix = distance_matrix
+
+    @classmethod
+    def calc_distance_matrix(cls, coordinates: List[Union[tuple, list]]) -> List[List[Union[int, float]]]:
+        """
+        Creates a distance matrix from a list of city coordinates.
+        :param coordinates: An iterable that contains tuples or lists representing a x,y coordinate.
+        :return: A list of lists containing the distances between cities.
+        """
+        res = [[0 for i in coordinates] for j in coordinates]
+        for i, coord_i in enumerate(coordinates):
+            for j, coord_j in enumerate(coordinates):
+                dist = math.sqrt(sum([(z[0] - z[1])**2 for z in zip(coord_i, coord_j)]))
+                res[i][j] = dist
+                res[j][i] = dist
+        return res
+
+    def check_solution(self, solution: List[int]):
+        """
+        Check if the solution for the TSP problem is valid.
+        :param solution: List of integers which refer to cities.
+        :return: None, unless errors are raised.
+        """
+        set_solution = set(solution)
+        set_problem = set(range(len(self.distance_matrix)))
+        if len(solution) > len(self.distance_matrix):
+            raise RuntimeError("Solution is longer than number of towns!")
```

Comment:
I'd say `ValueError` is more appropriate. Same on line 42.

Reply:
fixed. also fixed it in the santa part of the codebase.